CsvTextFieldParser/CsvTextFieldFormatter.cs

```diff
@@ -0,0 +1,293 @@
+                {
+                    if (ch == quoteChar)
+                    {
+                        writer.Write(quoteEscapeChar);
+                    }
+                    writer.Write(ch);
+                }
+                writer.Write(quoteChar);
+            }
+            else
+            {
+                writer.Write(field);
+            }
+        }
+
+        private void WriteEol()
+        {
+            if (eol != null)
+            {
+                writer.Write(eol);
+            }
+            else
+            {
+                writer.WriteLine();
+            }
+        }
+
+        private char[] SpecialChars
+        {
+            get
+            {
+                var specialChars = specialCharsCache;
+                if (specialChars == null)
+                {
+                    specialChars = new SortedSet<char> { delimiterChar, quoteChar, quoteEscapeChar, '\n', '\r' }.ToArray();
+                    specialCharsCache = specialChars;
+                }
+                return specialChars;
+            }
+        }
+
+        /// <summary>
+        /// Closes the current <see cref="CsvTextFieldFormatter"/> object.
+        /// </summary>
+        public void Close()
+        {
+            if (writer != null)
+            {
+                writer.Close();
+                writer = null;
+            }
+        }
+
+        /// <summary>
+        /// Closes and disposes the current <see cref="CsvTextFieldFormatter"/> object.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes of the current <see cref="CsvTextFieldFormatter"/> object.
+        /// </summary>
+        /// <param name="disposing">true if called from <see cref="Dispose()"/>, or false if called from a finalizer</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                Close();
+            }
+        }
+
+        #region Configuration
+
+        /// <summary>
+        /// Defines the delimiters for a text file.
+        /// Default is a comma.
+        /// </summary>
+        /// <remarks>
+        /// This is defined as an array of strings for compatibility with <code>Microsoft.VisualBasic.FileIO.TextFieldParser</code>,
+        /// but this formatter only supports one single-character delimiter.
+        /// </remarks>
+        /// <exception cref="ArgumentException">A delimiter value is set to a newline character, an empty string, or null.</exception>
+        /// <exception cref="NotSupportedException">The delimiters are set to an array that does not contain exactly one element with exactly one character.</exception>
+        public string[] Delimiters
+        {
+            get
+            {
+                return new string[] { delimiterChar.ToString(CultureInfo.InvariantCulture) };

```

Comment:
Why use ToString(CultureInfo.InvariantCulture) here?

Reply:
That's to satisfy the .NET Analyzer warning [CA1304: Specify CultureInfo](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1304).